code/modules/cooking/cooking_container.dm

```diff
@@ -0,0 +1,307 @@
+	if(tracker)
+		return PCWJ_CONTAINER_BUSY
+	if(length(contents) || reagents.total_volume > 0)
+		return PCWJ_CONTAINER_BUSY
+
+	return PCWJ_CONTAINER_AVAILABLE
+
+/obj/item/reagent_containers/cooking/item_interaction(mob/living/user, obj/item/used, list/modifiers)
+	if(!tracker && (contents.len || reagents.total_volume != 0))
+		to_chat(user, "The [src] is full. Empty its contents first.")
+		return ITEM_INTERACT_COMPLETE
+
+	process_item(user, used)
+
+	return ITEM_INTERACT_COMPLETE
+
+/obj/item/reagent_containers/cooking/proc/process_item(mob/user, obj/used)
+	if(!istype(used))
+		return PCWJ_NO_STEPS
+
+	if(!tracker)
+		if(!(type in GLOB.pcwj_recipe_dictionary))
+			return PCWJ_NO_STEPS
+
+		var/list/container_recipes = GLOB.pcwj_recipe_dictionary[type]
+		if(!length(container_recipes))
+			return PCWJ_NO_STEPS
+
+		tracker = new(src)
+
+		for(var/datum/cooking/recipe/recipe in container_recipes)
+			tracker.recipes_last_completed_step[recipe] = 0
+
+	react_to_process(tracker.process_item_wrap(user, used), user, used)
+
+/obj/item/reagent_containers/cooking/proc/react_to_process(reaction_status, mob/user, obj/used)
+	if(istype(used, /obj/machinery/cooking) && reaction_status == PCWJ_NO_STEPS)
+		// When a finished recipe is still sitting on a cooking machine, and the
+		// cooking machine is sending periodic process pings as it is activated,
+		// the container will still be processed and a tracker created, because
+		// the result from the finished recipe counts as the potential input to
+		// a new recipe. Which is valid; it may be the input to a new recipe.
+		// But if it's not, we don't want a message to keep showing up as if the
+		// player is trying to step through a recipe.
+		return
+
+	if(reaction_status == PCWJ_NO_STEPS && !tracker.recipe_started)
+		to_chat(user, "<span class='notice'>You don't know what you'd begin to make with this.</span>")
+		return
+
+	switch(reaction_status)
+		if(PCWJ_NO_RECIPES)
+			to_chat(user, "<span class='notice'>You don't know what you'd begin to make with this.</span>")
+		if(PCWJ_NO_STEPS)
+			to_chat(user, "<span class='notice'>You get a feeling this wouldn't improve the recipe.</span>")
+		if(PCWJ_SUCCESS)
+			if(tracker.step_reaction_message && ismob(user))
+				to_chat(user, "<span class='notice'>[tracker.step_reaction_message]</span>")
+
+			update_appearance(UPDATE_ICON)
+		if(PCWJ_COMPLETE)
+			if(tracker.step_reaction_message && ismob(user))
+				to_chat(user, "<span class='notice'>[tracker.step_reaction_message]</span>")
+				to_chat(user, "<span class='notice'>You finish cooking with \the [src].</span>")
+			QDEL_NULL(tracker)
+			clear_cooking_data()
+			update_appearance(UPDATE_ICON)
+
+/obj/item/reagent_containers/cooking/proc/handle_burning()
+	// Only create a burnt mess with objects in us, so boiled water doesn't
+	// turn into a burnt mess when left on the stove too long.
+	if(length(contents))
+		clear()
+		new/obj/item/food/badrecipe(src)
+	else
+		clear()
```

Comment:
```suggestion
	clear()
	if(length(contents))
		new/obj/item/food/badrecipe(src)
```

Reply:
The `clear()` call empties the contents, so if written like this the burnt mess would never be created.